### PR TITLE
Support standalone dvc installation via metapackages. 

### LINKED ISSUE #160: Install extras as separate packages?
This `dvc` package is (intentionally) heavily batteries-included and pulls in all the DVC extras as explicit requirements. This is great for zero-muss installations on conda, however it causes problems in environments where we may need to render a more minimal set of external dependencies.

Judging from DVC's [setup.py](https://github.com/iterative/dvc/blob/master/setup.py) and [Installation Instructions](https://dvc.org/doc/install/linux), the remote storage extras are *just* additional package requirements and don't modify the files packaged by DVC.

Would you be willing to refactor this package to allow a "no-extras" installation mode? This could be tackled through a multi-output build recipe that generates a `dvc` [metapackage](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#implicit-metapackages) for the top-level "all-remotes" build, with a `_dvc-core` primary packages and `_dvc-[extra]` packages for each remote extra.

### PR DESCRIPTION
**Pull request by asford**
_closes #160_
_adds @asford to maintainers to support metapackage layout_ 

Build a `dvc` conda package equivalent to the `dvc[all]` pip
package, with a `_dvc` base package containing dvc alone. This allows a
happy-path of:

  `conda install -c conda-forge dvc`

supporting dvc and all remotes and a power-user path of:

  `conda install -c conda-forge _dvc _dvc-s3`

for dvc and specific remotes.

See comments in meta.yml for details.

-----------------------

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by efiop** [comment]
Thank you @asford ! Looks neat! :sparkles: 

Btw, looks like windows is failing, could you take a look, please?

**Comment by asford** [comment]
@conda-forge-admin, please rerender

**Comment by asford** [comment]
@conda-forge-admin, please rerender

**Comment by asford** [comment]
@conda-forge-admin, please rerender

**Comment by asford** [comment]
@efiop I've gone ahead and fixed the windows build and polished this up for the smithy and linter. Would you mind reviewing the `meta.yaml` to ensure that you understand the core logic?

The interaction between the smithy/linter/build was a *little* persnickety, as there are multiple equivalent ways of phrasing this for conda-build but only one or two that dual-satisfy the linter and smithy.  I've added myself to the maintainers list so I can proactively reach out with any issues due to the metapackage layout. Feel free to revert bfb745229b74e4715fbada221e2e394aebfb4adb if you'd prefer to keep the maintainers list short.

**Review by efiop — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,70 +11,179 @@ source:
...
     - build.patch
 
+# The "true" build is specified in the _dvc output below.
+# Of the build properties here and below:
+# noarch is not inherited and must be specified per output
+# number is propagated into the outputs below
 build:
-  number: 0
-  script: {{ PYTHON }} -m pip install . -vv
-  skip: true  # [py<36]
```
Comment (efiop) [inline recipe/meta.yaml:17]: Did we lose this?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,70 +11,179 @@ source:
...
+#   `conda install -c conda-forge _dvc _dvc-s3`
+#
+# for dvc and specific remotes.
+#
+# Here's how it works...
+#
+# We define extras through "private" subpackages and a single top-level `dvc`
+# metapackage. The metapackage at a given build version references the extras
+# subpackages at the same build version. `_dvc` includes the dvc package itself
+# and core dependencies. `_dvc-gs`, `_dv-s3`, et al contain no code, but
```
Comment (efiop) [inline recipe/meta.yaml:49]:
```suggestion
# and core dependencies. `_dvc-gs`, `_dvc-s3`, et al contain no code, but
```

**Comment by efiop** [comment]
@asford Thank you so much! Looks great! Just a few minor comments above.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,70 +11,179 @@ source:
   patches:
     - build.patch
 
+# The "true" build is specified in the _dvc output below.
+# Of the build properties here and below:
+# noarch is not inherited and must be specified per output
+# number is propagated into the outputs below
 build:
```
Comment (asford) [inline recipe/meta.yaml:18]:
```suggestion
build:
  skip: true # [py<36]
```

**Comment by asford** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (asford) [reply recipe/meta.yaml:17]: I *think* conda-forge doesn't target `py<36` anymore, so we weren't seeing a delta in the build matrix, but I've reintroduced for the sake of clarity in the recipe.

**Comment by asford** [comment]
@efiop Thanks for the catch on the review. This should be good-to-merge pending final build passes.

**Comment by efiop** [comment]
Thank you @asford ! :pray:

**Pull request merged**